### ISSUE TITLE
SfM reconstruction flipped w.r.t Z axis

### ISSUE THREAD

**Issue by fayoub** [openMVG/openMVG#2258]
Hello,
I've run a series of SfM processes using Incremental, IncrementalV2, Global with various options, with GPS prior. The reconstruction looks good except it's always flipped/mirrored with respect to the Z axis. See screenshot of a bit of the reconstruction. 
<img width="893" alt="Screenshot 2023-10-24 at 10 54 49 PM" src="https://github.com/openMVG/openMVG/assets/23509694/becf5d57-e72c-494c-a7b1-1d4af5c59c08">
It's a drone flight take off. The GPS prior (blue) are correct, but the reconstructed camera positions (green) are flipped. The problem with that is that I can't reorient the reconstruction with a 7dof similarity, because of the flip.
Does anybody knows why is the solution consistently flipped? And what could be a way to prevent this from happening?
Thank you and congrats on this great tool!

**Comment by pmoulon** [comment]
Could be that there is a ambiguity if all your GPS prior are in the same 2D plane?
- are you GPS prior on a plane?
- try with more distributed GPS (that would not fit in a 2D plane)

**Comment by fayoub** [comment]
Thanks Pierre for the reply.
That's the full acquisition:
![Screenshot 2023-10-25 at 6 02 23 PM](https://github.com/openMVG/openMVG/assets/23509694/a3bc1963-67ca-4333-bb52-0753f5290e1f)
The GPS priors are not on a plane. Note that w.r.t the prior, it's custom code, where the priors are directly expressed in a right-handed local reference frame with Z pointing "down" (Z decreases as the drone takes off). I've tried reverting the coordinates system with Z up, but that didn't make a difference. Prior and reconstruction are flipped.
I understand why there could be an ambiguity, but what puzzles me is that no matter what kind of SfM engine I use or initial pair I use, the reconstruction is always flipped. I would have expected a some randomness between flipped vs non-flipped reconstruction if it's an ambiguity. 
I've tried Incremental engine with initialization from the location where the craft turns (more random movements) - see pic - but still, reconstruction is flipped.
![Screenshot 2023-10-25 at 6 17 00 PM](https://github.com/openMVG/openMVG/assets/23509694/b086af51-832d-48a5-b7e4-ed3f4803cfc6)
